### PR TITLE
feat(sdk-core): add getDerivationPath method for smc wallets

### PR DESCRIPTION
**Pull request by alia-bitgo**
TICKET: WP-521

### CODE REVIEW

File: modules/sdk-core/src/bitgo/tss/eddsa/utils.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import crypto from 'crypto';
+
+/**
+ * @param seed - used to construct derivation path deterministically
+ * @param isMaster - if set, path starts with prefix `m/`
+ * @return path `(m/)/999999/a/b` where `a` and `b` are 7-byte pseudorandom numbers based on seed
+ */
+export function getDerivationPath(seed: string, isMaster = true): string {
```
Comment (zahin-mohammad) [inline modules/sdk-core/src/bitgo/tss/eddsa/utils.ts:8]: Nothing here seems eddsa specific, can this be a generic util instead? Or conversely can we verify that the logic is the same or different with ecdsa?

File: modules/sdk-core/src/bitgo/tss/utils.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import crypto from 'crypto';
+
+/**
+ * @param seed - used to construct derivation path deterministically
+ * @param isMaster - if set, path starts with prefix `m/`
+ * @return path `(m/)/999999/a/b` where `a` and `b` are 7-byte pseudorandom numbers based on seed
+ */
+export function getDerivationPath(seed: string, isMaster = true): string {
```
Comment (zahin-mohammad) [inline modules/sdk-core/src/bitgo/tss/utils.ts:8]: Please add tests for this function.

**Review by zahin-mohammad — CHANGES_REQUESTED** [review]

File: modules/sdk-core/src/bitgo/tss/eddsa/utils.ts (hunk 1 above)
Comment (zahin-mohammad) [reply modules/sdk-core/src/bitgo/tss/eddsa/utils.ts]: Just want to avoid devs re-creating this function and introducing differences in logic.

File: modules/sdk-lib-mpc/src/util.ts [hunk 3]
```diff
@@ -137,3 +138,22 @@ export async function randomPositiveCoPrimeLessThan(x: bigint): Promise<bigint>
...
+
+/**
+ * @param seed - used to construct derivation path deterministically
+ * @param isMaster - if set, path starts with prefix `m/`
+ * @return path `(m/)/999999/a/b` where `a` and `b` are 7-byte pseudorandom numbers based on seed
+ */
+export function getDerivationPath(seed: string, isMaster = true): string {
+  function sha256(input: crypto.BinaryLike): Buffer {
+    return crypto.createHash('sha256').update(input).digest();
+  }
```
Comment (zahin-mohammad) [inline modules/sdk-lib-mpc/src/util.ts:150]: Can we move this function outside? There is no reason to re-create this function every time we call `getDerivationPath`

Comment (zahin-mohammad) [reply modules/sdk-lib-mpc/src/util.ts:150]: It can remain a a private/un-exported function, but define outside of `getDerivationPath`

File: modules/sdk-lib-mpc/test/unit/util.ts [hunk 4]
```diff
@@ -75,4 +81,10 @@ describe('mpc utils', () => {
       }
     });
   });
+
+  it('should get derivation path from a provided seed', function () {
+    const seed = '1';
+    const derivationPath = getDerivationPath(seed);
+    assert.strictEqual(derivationPath, 'm/999999/163767512/266960264');
+  });
```
Comment (zahin-mohammad) [inline modules/sdk-lib-mpc/test/unit/util.ts:89]: Missing tests with `isMaster=false`. 

Also missing tests around some edge cases. What happens when `seed` is an empty string?

**Review by zahin-mohammad — CHANGES_REQUESTED** [review]

**Review by zahin-mohammad — APPROVED** [review]
